File: converted/source2/source/zombie/characters/action/conditions/EventNotOccurred.java
package zombie.characters.action.conditions;

import org.w3c.dom.Element;
import zombie.characters.action.ActionContext;
import zombie.characters.action.IActionCondition;

public final class EventNotOccurred implements IActionCondition {
   public String eventName;

   public String getDescription() {
      return "EventNotOccurred(" + this.eventName + ")";
   }

   private boolean load(Element var1) {
      this.eventName = var1.getTextContent().toLowerCase();
      return true;
   }

   public boolean passes(ActionContext var1, int var2) {
      return !var1.hasEventOccurred(this.eventName, var2);
   }

   public IActionCondition clone() {
      return null;
   }
}

File: ProjectZomboidSwitch/src/zombie/characters/action/conditions/EventNotOccurred.cpp
#include <string>
#include "zombie\characters\action\conditions/EventNotOccurred.h"

namespace zombie {
namespace characters {
namespace action {
namespace conditions {

std::string EventNotOccurred::getDescription() {
    // TODO: Implement getDescription
    return "";
}

bool EventNotOccurred::load(Element var1) {
    // TODO: Implement load
    return false;
}

bool EventNotOccurred::passes(ActionContext var1, int var2) {
    // TODO: Implement passes
    return false;
}

IActionCondition EventNotOccurred::clone() {
    // TODO: Implement clone
    return nullptr;
}

} // namespace conditions
} // namespace action
} // namespace characters
} // namespace zombie
